discovery-infra/test_infra/helper_classes/infra_env.py

```diff
@@ -0,0 +1,97 @@
+import os
+import logging
+from typing import Optional
+from junit_report import JunitTestCase
+from test_infra import consts, utils
+from test_infra.assisted_service_api import InventoryClient
+from test_infra.helper_classes.config import BaseInfraEnvConfig
+from test_infra.helper_classes.nodes import Nodes
+from test_infra.tools import static_network, terraform_utils
+
+
+
+class InfraEnv:
+    def __init__(self, api_client: InventoryClient, config: BaseInfraEnvConfig, nodes: Optional[Nodes] = None):
+        self._config = config
+        self.nodes = nodes
+        self.api_client = api_client
+        try:
+            infra_env = self._create()
+        except:
+            logging.exception("create")
+            raise
+        self._config.infra_env_id = self.id = infra_env.id
+
+    def _create(self):
+        return self.api_client.create_infra_env(
+            self._config.infra_env_name.get(),
+            pull_secret=self._config.pull_secret,
+            ssh_public_key=self._config.ssh_public_key,
+            openshift_version=self._config.openshift_version,
+        )
+
+    @property
+    def config(self):
+        return self._config.get_copy()
+
+    def update_config(self, **kwargs):
+        """
+        Note that kwargs can contain values for overriding BaseInfraEnvConfig arguments.
+        The name (key) of each argument must match to one of the BaseInfraEnvConfig arguments.
+        If key doesn't exists in config - KeyError exception is raised
+        """
+        logging.info(f"Updating infra-env {self.id} configurations to {kwargs}")
+
+        for k, v in kwargs.items():
+            if not hasattr(self._config, k):
+                raise KeyError(f"The key {k} is not present in {self._config.__class__.__name__}")
+            setattr(self._config, k, v)
+
+    def prepare_infraenv(self, static_network_config=None, **kwargs):
+        self.update_config(**kwargs)
+        logging.info(f"Preparing for installation with infra-env configurations: infraenv_config={self._config}")
+        self.nodes.controller.log_configuration()
+        if self._config.download_image:
+            if static_network_config:
+                static_network_config = static_network. \
+                    generate_static_network_data_from_tf(self.nodes.controller.tf_folder)
+            else:
+                static_network_config = None
```

Comment:
where static_network_config is passed to infra-env?

Reply:
fixed